src/uri.rs

```diff
@@ -389,6 +389,48 @@ impl StdError for UriError {
     }
 }
 
+// TODO: Is ByteStr adding value? Inline it?
```

Comment:
Not a rebuttal, I'm just curious what you mean by shared. Doesn't `ByteStr` owns `Bytes`?